### ISSUE TITLE
Installation on linux question

### ISSUE THREAD

**Issue by davidcraft** [hklarner/PyBoolNet#16]
After installing, I run python and type 
>> import PyBoolNet
and get
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "PyBoolNet/__init__.py", line 6, in <module>
    import PyBoolNet.AttractorBasins
  File "PyBoolNet/AttractorBasins.py", line 15, in <module>
    import PyBoolNet.FileExchange
  File "PyBoolNet/FileExchange.py", line 9, in <module>
    import PyBoolNet.PrimeImplicants
  File "PyBoolNet/PrimeImplicants.py", line 4, in <module>
    import PyBoolNet.QuineMcCluskey
  File "PyBoolNet/QuineMcCluskey.py", line 12, in <module>
    with open(fname_nusmvkeywords) as f:
IOError: [Errno 2] No such file or directory: 'PyBoolNet/Dependencies/nusmvkeywords.json'

I'm not a python user... I'm sure there is a simple fix for this! Any guidance? 

Also, in general do you provide a full example file which reads an sbml file and returns a steady state?

EDIT: @davidcraft It looks like the problem is that you were trying to install the github repository of PyBoolnet rather than the actual release file, see new comments below and the linux installation section 1.2 in the PyBoolNet manual which is currently [PyBoolNet-2.2.0_manual.pdf](https://github.com/hklarner/PyBoolNet/releases/download/v2.2.0/PyBoolNet-2.2.0_manual.pdf). The latest manual is always available at https://github.com/hklarner/PyBoolNet/releases.

In a nutshell, download the tar.gz file that agrees with your os from https://github.com/hklarner/PyBoolNet/releases and install with pip. For example: `sudo pip install PyBoolNet-2.2.0_linux64.tar.gz`

**Comment by hklarner** [comment]
hi David,
Regarding the installation: I need more information. Which PyBoolNet file did you download and install? Are you using Python 2 or 3?

Regarding smbl: I haven't implemented my own SBML parser as it is fairly elaborate. I suggest you either use the R package [BoolNet](https://cran.r-project.org/web/packages/BoolNet/index.html) and in particular the functions `loadSBML(..)` and `save_network(..)`. Example R code to convert `mapk.sbml` to `mapk.bnet`:

```
library(BoolNet)
net = loadSBML("mapk.sbml")
saveNetwork(net, "mapk.bnet")
```

Alternatively you can import your sbml file into the software [GINsim](http://ginsim.org/home) and save it as a bnet file. To do so I suggest to download the [nightly build](http://ginsim.org/downloads) of GINsim.

**Comment by burgerga** [comment]
This happens if you download and install the git repository, eg by executing

```
pip install git+https://github.com/hklarner/PyBoolNet@v2.2.0
```
(would be really awesome if this worked by the way ;) )

The solution is to download the appropriate version from https://github.com/hklarner/PyBoolNet/releases 
Maybe the installation approach can be added to the README?

**Comment by hklarner** [comment]
Very good point. There seems to be continuous confusion about how to install PyBoolNet. The preferred way is indeed to download the latest version from  https://github.com/hklarner/PyBoolNet/releases and then install with pip. I will add this info to the README.

I will look into making
```
pip install git+https://github.com/hklarner/PyBoolNet@v2.2.0
```
possible.

**Comment by carissableker** [comment]
Perhaps something like this added to the setup.py would help: 

```python
import platform

this_os = platform.system()
if this_os == "Linux":
    dep_folder = os.path.join("Dependencies", "linux64") 
elif this_os == "Darwin":
    dep_folder = os.path.join("Dependencies", "mac64") 
elif this_os == "Windows":
    dep_folder = os.path.join("Dependencies", "win64") 
else:
    # no idea if we could get here?
    raise Exception 

# copy dependencies to PyBoolNet/Dependencies
from distutils.dir_util import copy_tree

copy_tree(dep_folder, os.path.join("PyBoolNet", "Dependencies")

```

**Comment by hklarner** [comment]
It would be great to get
```
pip install git+https://github.com/hklarner/PyBoolNet@v2.2.0
```
to work. @carissableker is your suggestion a step in that direction?

**Comment by carissableker** [comment]
@hklarner 
Yes, the only part of [setup.py](https://github.com/hklarner/PyBoolNet/blob/master/setup.py) that I can see as issue preventing that to work, is that it does not copy the correct (OS dependent) dependencies from `Dependencies` to `PyBoolNet/Dependencies`. This should fix that, and I tested it on my fork (https://github.com/carissableker/PyBoolNet):

```bash
$ conda create -n test-env -y python
[...]
$ conda activate test-env
(test-env) $ pip install git+https://github.com/carissableker/PyBoolNet
Collecting git+https://github.com/carissableker/PyBoolNet
  Cloning https://github.com/carissableker/PyBoolNet to /tmp/pip-req-build-si7az139
  Running command git clone -q https://github.com/carissableker/PyBoolNet /tmp/pip-req-build-si7az139
Collecting networkx==1.11
  Using cached networkx-1.11-py2.py3-none-any.whl (1.3 MB)
Collecting decorator>=3.4.0
  Using cached decorator-4.4.2-py2.py3-none-any.whl (9.2 kB)
Building wheels for collected packages: PyBoolNet
  Building wheel for PyBoolNet (setup.py) ... done
  Created wheel for PyBoolNet: filename=PyBoolNet-2.2.8a0-py3-none-any.whl size=7073883 sha256=c50d01821b2f6bc51b49b6ad96e587e86ee6bc79332970154f8c7e419fc0e8f7
  Stored in directory: /tmp/pip-ephem-wheel-cache-r5lpdapn/wheels/0e/72/a3/16af3854ac5db3c9f53318f280f201a7d1bcead0c8b3aa93e4
Successfully built PyBoolNet
Installing collected packages: decorator, networkx, PyBoolNet
Successfully installed PyBoolNet-2.2.8a0 decorator-4.4.2 networkx-1.11
(test-env) $ python -c "import PyBoolNet"
(test-env) $ 
```

I am not sure how it affects [make_release.sh](https://github.com/hklarner/PyBoolNet/blob/master/make_release.sh) and I have only tested it on Linux (Ubunte Budgie). 

More testing the install: 
![image](https://user-images.githubusercontent.com/26261557/86339853-5e160e80-bc54-11ea-99e2-fd76d2132d91.png)

Probably irrelevant, but `PyBoolNet.Tests.run()` fails because of  [Issue 23](https://github.com/hklarner/PyBoolNet/issues/23) and I do not want to use `sudo` to fix it as it does not affect my usage.

**Comment by hklarner** [comment]
It works for me too. The `make_realse.sh` would be replaced by this way of installing PyBoolNet. @aurelien-naldi what do you think? Does this interfere with how you are using PyBoolNet? (BTW: do you know each other?) Otherwise I would say, @carissableker open a PR and I will merge.

**Comment by carissableker** [comment]
Made a pull request (my first one ever!). 

I do not know @aurelien-naldi , but I looked at his website and it looks like his publishments and tools will help with the  work I have just started.  Thanks! :smile:

**Comment by aurelien-naldi** [comment]
It should not interfere with what I want to do with the conda packages. To build the conda package, I should be able to call the same "pip install" command and simply remove the dependencies, as they would be provided by other conda packages

**Comment by aurelien-naldi** [comment]
Thinking a bit more about it, it would be even better for me if setup.py could first check if the CONDA_BUILD environment variable is set to skip the copy of dependencies when it is, instead of removing them afterwards.

Once these changes are in, I will add a pull request to include the conda recipe
